diff --git a/configure b/configure
index d2e7a89..d8774ee 100755
--- a/configure
+++ b/configure
@@ -5700,6 +5700,10 @@ case "$target" in
                 TARGET_ARCH="linux64"
                 CPU_ARCH=64
                 ;;
+            aarch64)
+                TARGET_ARCH="linux64"
+                CPU_ARCH=64
+                ;;
             i*86)
                 TARGET_ARCH="linux"
                 CPU_ARCH=32
